fluentui_progress/src/main/java/com/microsoft/fluentui/tokenized/progress/ProgressText.kt

```diff
@@ -0,0 +1,138 @@
+    progressTextTokens: ProgressTextTokens? = null
+) {
+    val tokens = progressTextTokens
+        ?: FluentTheme.controlTokens.tokens[ControlTokens.ControlType.ProgressText] as ProgressTextTokens
+    val progressTextInfo = ProgressTextInfo()
+    val currentProgress = animateFloatAsState(
+        targetValue = progress.coerceIn(0f..1f), animationSpec = tween(
+            delayMillis = 0, durationMillis = 1000, easing = LinearOutSlowInEasing
+        )
+    )
+    val backgroundBrush = tokens.backgroundBrush(progressTextInfo = progressTextInfo)
+    val border = tokens.borderWidth(progressTextInfo = progressTextInfo)
+    val borderColor = tokens.borderColor(progressTextInfo = progressTextInfo)
+    val typography = tokens.typography(progressTextInfo = progressTextInfo)
+    val textColor = tokens.textColor(progressTextInfo = progressTextInfo)
+    val progressIndicatorHeight = tokens.progressbarHeight(progressTextInfo = progressTextInfo)
+    val progressIndicatorBackgroundColor =
+        tokens.progressbarBackgroundColor(progressTextInfo = progressTextInfo)
+    val progressIndicatorColor = tokens.progressbarBrush(progressTextInfo = progressTextInfo)
+    val iconTextSpacing = tokens.iconTextSpacing(progressTextInfo = progressTextInfo)
+    val padding = tokens.padding(progressTextInfo = progressTextInfo)
+    val iconSize = tokens.iconSize(progressTextInfo = progressTextInfo)
+    val iconColor = tokens.iconColor(progressTextInfo = progressTextInfo)
+    val shape = RoundedCornerShape(12.dp)
+    Box(modifier = Modifier.clip(shape)) {
+        Box(
+            modifier = modifier
+                .background(
+                    backgroundBrush, shape
+                )
+                .border(
+                    border, borderColor, shape
+                )
+        ) {
+            Row(
+                modifier = Modifier.padding(padding),
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                Icon(
+                    ProgressTextIcons.DismissCircle,
+                    contentDescription = LocalContext.current.resources.getString(R.string.fluentui_cancel),
+                    modifier = Modifier
+                        .size(iconSize)
+                        .clickable(
+                            interactionSource = remember { MutableInteractionSource() },
+                            indication = LocalIndication.current,
+                            enabled = true,
+                            onClickLabel = null,
+                            role = Role.Button,
+                            onClick = onCancelClick
+                        ),
+                    tint = iconColor
+                )
+                Spacer(modifier = Modifier.width(iconTextSpacing))
+                BasicText(text = text, style = typography.merge(TextStyle(color = textColor)))
+            }
+        }
+        Box(modifier = Modifier.align(Alignment.BottomCenter)) {
```

Comment:
2 boxes in a Box with alignments set to orient them. Can we use Column or something? Any reason this hierarchy is used?

Reply:
If we use Column, two boxes are set one below the other but the problem is we want the 2nd box to be on the bottom border of the parent box, so box is used with orientation bottomcenter